### PR TITLE
Restore MartenOps.Store and add MartenOps.StoreMany

### LINKED ISSUE #1162: Store multiple documents with StoreDoc bug
**Describe the bug**
In the Wolverine [V3.4.0](https://github.com/JasperFx/wolverine/releases/tag/V3.4.0) `StoreDoc<T>` now stores an `object[]` with one element of type `T[]`.

**To Reproduce**
```
public recod Foo;

public void MartenOps_store_test()
{
    var docs = new[] { new Foo(), new Foo() };

    var storeOp = MartenOps.Store<Foo>(docs);

    storeOp.Documents.ShouldBe(docs);
}

```

**Expected behavior**
`StoreDoc<T>` should store an `object[]` with elements of type `T`.

**Additional context**
- The bug was introduced in [this commit](https://github.com/JasperFx/wolverine/commit/e3906fd026596c8bd6c73bb71b9002ae1182498d).
- The initial change from [this commit](https://github.com/JasperFx/wolverine/commit/c1eec8ec71c0ba8095bf87503c9a3f77fef3c126) introduced a breaking change in a minor version.

### PR DESCRIPTION
**Pull request by jay-zahiri**
Fixes #1162

### CODE REVIEW

File: src/Persistence/Wolverine.Marten/IMartenOp.cs [hunk 1]
```diff
@@ -225,10 +242,25 @@ public void Execute(IDocumentSession session)
...
+        session.Store(_document);
+    }
+}
+
+public class StoreManyDocs<T> : DocumentsOp where T : notnull
 {
     private readonly T[] _documents;
 
-    public StoreDoc(params T[] documents) : base(documents)
+    public StoreManyDocs(T[] documents) : base(documents)
```
Comment (davidst) [inline src/Persistence/Wolverine.Marten/IMartenOp.cs:263]: https://github.com/JasperFx/wolverine/issues/1162#issuecomment-2530791285

Comment (jay-zahiri) [reply src/Persistence/Wolverine.Marten/IMartenOp.cs]: Missed that! Fixed it

File: src/Persistence/Wolverine.Marten/IMartenOp.cs [hunk 2]
```diff
@@ -285,12 +317,24 @@ public override void Execute(IDocumentSession session)
...
+
+    public abstract void Execute(IDocumentSession session);
+}
+
+public abstract class DocumentsOp : IMartenOp
 {
     public object[] Documents { get; }
 
-    protected DocumentOp(params object[] documents)
+    protected DocumentsOp(IEnumerable<object> documents)
```
Comment (davidst) [inline src/Persistence/Wolverine.Marten/IMartenOp.cs:335]: I suggest keeping the `params` keyword to make the usage nicer.

```suggestion
    protected DocumentsOp(params object[] documents)
```

File: src/Persistence/Wolverine.Marten/IMartenOp.cs [hunk 3]
```diff
@@ -225,12 +242,27 @@ public void Execute(IDocumentSession session)
...
+        session.Store(_document);
+    }
+}
+
+public class StoreManyDocs<T> : DocumentsOp where T : notnull
 {
     private readonly T[] _documents;
 
-    public StoreDoc(params T[] documents) : base(documents)
+    public StoreManyDocs(IEnumerable<T> documents) : base(documents.Cast<object>())
```
Comment (davidst) [inline src/Persistence/Wolverine.Marten/IMartenOp.cs:263]: I suggest keeping the `params` keyword to make the usage nicer.

```suggestion
    public StoreManyDocs(params T[] documents) : base(documents.Cast<object>().ToArray())
```

**Review by davidst — CHANGES_REQUESTED** [review]

**Comment by davidst** [comment]
I suggest to also add some unit tests to verify the correct behavior.

```
public recod Foo;

public void StoreManyDocs_test()
{
    var docs = new[] { new Foo(), new Foo() };

    var storeOp = MartenOps.StoreManyDocs<Foo>(docs);

    storeOp.Documents.ShouldBe(docs);
}
```

File: src/Persistence/Wolverine.Marten/IMartenOp.cs (hunk 3 above)
Comment (jay-zahiri) [reply src/Persistence/Wolverine.Marten/IMartenOp.cs]: Ok, I added two overload methods to this to align with other `params` methods in `MartenOps`, to keep a consistent DX.

**Review by davidst — APPROVED** [review]

**Comment by jay-zahiri** [comment]
I think it's a bit overkill to test the language itself, but you're welcome to add that. [Here](https://dotnetfiddle.net/ah4hXJ) is a quick update on the demo, if you're unsure.

**Comment by davidst** [comment]
I agree that it is overkill to test the language itself, but a test for this has nothing to do with testing the language itself. A bug sneaked into the implementation, therefore a test to verify the correct implementation of the `StoreManyDocs` would be appropriate.

**Pull request merged**